fake.go

```diff
@@ -519,6 +520,80 @@ func (fake *Fake) Dump() string {
 	return buf.String()
 }
 
+var commentRegexp = regexp.MustCompile(`^(.*)comment "([^"]*)"( ;)?([^"]*)$`)
+
+// splitComment splits line into a required pre-comment and optional trailing comment
+// (which is enclosed in quotes but does not contain any quotes).
+func splitComment(line string) (string, *string) {
+	//fmt.Printf("Split comment !%v!\n", line)
+	// We could perhaps do this more efficiently without using a regexp, but it would
+	// be more complicated...
+	match := commentRegexp.FindStringSubmatch(line)
+	if match != nil {
+		return strings.TrimSpace(match[1]) + match[4], &match[2]
+	}
+	return line, nil
+}
+
+// ParseDump can parse a dump for a given nft instance.
+// It expects table name and the same nftables family in all rules based on a given fake.nftContext.
+// The best way to verify that everything important was properly parsed is to
+// compare given data with nft.Dump() output.
+func (fake *Fake) ParseDump(data string) (err error) {
+	lines := strings.Split(data, "\n")
+	var i int
+	var line string
+	parsingDone := false
+	defer func() {
+		if err != nil && !parsingDone {
+			err = fmt.Errorf("line %v: %w", i+1, err)
+		}
+	}()
+	txn := fake.NewTransaction()
+	for i, line = range lines {
+		line = strings.TrimSpace(line)
+		if line == "" || line[0] == '#' {
+			continue
+		}
+		words := strings.Split(line, " ")
+		if len(words) < 4 {
+			return fmt.Errorf("every command should have at least 4 words")
+		}
+		if words[0] != "add" {
+			return fmt.Errorf("only supports add operation, got %s", words[0])
+		}
+		if words[2] != string(fake.nftContext.family) {
+			return fmt.Errorf("wrong ip family %s, expected %s", words[2], string(fake.nftContext.family))
+		}
+		if words[3] != fake.nftContext.table {
+			return fmt.Errorf("wrong table name %s, expected %s", words[3], fake.nftContext.table)
+		}
+		var obj Object
+		switch words[1] {
+		case "table":
+			obj, err = parseTable(words)
+		case "chain":
+			obj, err = parseChain(words)
+		case "rule":
+			obj, err = parseRule(words)
+		case "map":
+			obj, err = parseMap(words)
+		case "set":
+			obj, err = parseSet(words)
+		case "element":
+			obj, err = parseElement(words)
+		default:
+			return fmt.Errorf("unknown object %s", words[1])
+		}
+		if err != nil {
+			return err
+		}
+		txn.Add(obj)
+	}
+	parsingDone = true
+	return fake.Run(context.TODO(), txn)
```

Comment:
Oh, don't do that; you see `context.TODO()` a lot in kubernetes and ovn-k because so much of the code was written before people started bothering with contexts and so lots of functions don't have a context to pass down. But we're not doing that here; if ParseDump() needs to have a context at this point, then you should make it take a context as an argument.

Reply:
yeah, that makes sense